spec/signature/types.go

```diff
@@ -0,0 +1,108 @@
+package signature
+
+import "crypto"
+
+const (
+	// MediaTypeDescriptor describes the media type of the descriptor.
+	MediaTypeDescriptor = "application/vnd.oci.descriptor.v1+json"
+)
+
+// Descriptor describes the content signed or to be signed.
+type Descriptor struct {
+	// The media type of the targeted content.
+	MediaType string `json:"mediaType"`
+
+	// The digest of the targeted content.
+	Digest string `json:"digest"`
+
+	// Specifies the size in bytes of the blob.
+	Size int64 `json:"size"`
+
+	// Contains optional user defined attributes.
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Equal reports whether d and t points to the same content.
+func (d Descriptor) Equal(t Descriptor) bool {
+	return d.MediaType == t.MediaType && d.Digest == t.Digest && d.Size == t.Size
+}
+
+// KeyType defines a key type and size.
+type KeyType string
```

Comment:
I'd rename to KeySpec to be inline with plugin spec, otherwise we have code like `KeySpec signature.KeyType `json:"keySpec"` where it gets confusing. Key type by itself usually means RSA, EC etc.

Reply:
Makes sense.